### PR TITLE
Clean cargo warnings

### PR DESCRIPTION
**Pull request by Ifropc**
### What

Cleanup cargo warnings

### Why

Cleanup cargo warnings

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-cli/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ default-run = "soroban"
 
 [[bin]]
 name = "stellar"
-path = "src/bin/main.rs"
+path = "../stellar-cli/src/bin/main.rs"
```
Comment (Ifropc) [inline cmd/soroban-cli/Cargo.toml:17]: Not sure if it's the cleanest way, but this way we are ensuring that we are pointing to a correct binary (doubt we would want to change soroban-cli and stellar-cli binaries in the future)

Comment (leighmcculloch) [reply cmd/soroban-cli/Cargo.toml:17]: We should ignore this warning. It's fine to have two binaries reference the same main. The warning is just to catch people me doing a mistake. We are doing it intentionally.

Comment (leighmcculloch) [reply cmd/soroban-cli/Cargo.toml:17]: There should be an option we can add to the projects `.cargo/config.toml` file or the the GitHub ci workflow to disable that warning.

**Review by leighmcculloch — COMMENTED** [review]
Looks good except we don't need to change the binary main references. Can you remove that change then it's good.

File: cmd/soroban-cli/Cargo.toml (hunk 1 above)
Comment (Ifropc) [reply cmd/soroban-cli/Cargo.toml:17]: Should we maybe just copy main to silence this warning?

Comment (Ifropc) [reply cmd/soroban-cli/Cargo.toml:17]: There's only `RUSTFLAGS=-Awarnings` which silences all warnings, which is not ideal. I'd rather either copy file (done in the last commit), or just keep the warning there

Comment (leighmcculloch) [reply cmd/soroban-cli/Cargo.toml]: > Should we maybe just copy main to silence this warning?

Yeah, that would be preferred over referencing the other crate file which is technically outside the project.

The files are small, couple copies is fine.

File: cmd/soroban-cli/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ path = "src/bin/main.rs"
```
Comment (leighmcculloch) [inline cmd/soroban-cli/Cargo.toml:18]: Could you name the other file `stellar.rs` for symmetry?

**Review by leighmcculloch — APPROVED** [review]
One ask to rename the other file to stellar.rs, but regardless lgtm. 👍🏻

**Pull request merged**